File: I/225-implement-stack-using-queues/main.go
package main

import "fmt"

type MyStack struct {
	Q []int
}

/** Initialize your data structure here. */
func Constructor() MyStack {
	return MyStack{Q: []int{}}
}

/** Push element x onto stack. */
func (this *MyStack) Push(x int) {
	this.Q = append(this.Q, x)
}

/** Removes the element on top of the stack and returns that element. */
func (this *MyStack) Pop() int {
	v := this.Q[len(this.Q)-1]
	this.Q = this.Q[0 : len(this.Q)-1]
	return v
}

/** Get the top element. */
func (this *MyStack) Top() int {
	return this.Q[len(this.Q)-1]
}

/** Returns whether the stack is empty. */
func (this *MyStack) Empty() bool {
	return len(this.Q) == 0
}

func main() {
	fmt.Println("a")
}

File: I/225-implement-stack-using-queues/main.rs
struct MyStack {
  q: Vec<i32>,
  c: usize,
}

impl MyStack {
  fn new() -> Self {
    MyStack {
      q: Vec::new(),
      c: 0,
    }
  }

  fn push(&mut self, x: i32) {
    if self.c >= self.q.len() {
      self.q.push(x);
      self.c = self.q.len();
    } else {
      self.q[self.c] = x;
      self.c+=1;
    }
  }

  fn pop(&mut self) -> i32 {
    self.c -= 1;
    self.q[self.c]
  }

  fn top(&self) -> i32 {
    self.q[self.c - 1]
  }

  fn empty(&self) -> bool {
    self.c == 0
  }
}

fn main() {
  let mut s: MyStack = MyStack::new();
  s.push(1);
  println!("{}", s.pop());
  s.push(2);
  println!("{}", s.top());
}
